<?xml version="1.0"?>
<!--
Copyright (C) 2017, Raffaello Bonghi <raffaello@rnext.it>
All rights reserved

Redistribution and use in source and binary forms, with or without
modification, are permitted provided that the following conditions are met:

1. Redistributions of source code must retain the above copyright 
   notice, this list of conditions and the following disclaimer.
2. Redistributions in binary form must reproduce the above copyright
   notice, this list of conditions and the following disclaimer in the
   documentation and/or other materials provided with the distribution.
3. Neither the name of the copyright holder nor the names of its 
   contributors may be used to endorse or promote products derived 
   from this software without specific prior written permission.
 
THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND 
CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, 
BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS 
FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, 
SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; 
OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, 
WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE 
OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, 
EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
-->
<launch>
  <!-- Set Lidar port -->
  <arg name="port" default="/dev/ttyUSB1" />

  <!-- Launch  imu node  -->
  <node pkg="ultron_kernel" type="mpu9250_serial_to_imu_node" name="imu_node" required="true" output="screen">
    <!-- load serial port   -->
    <param name="port"         type="string" value="$(arg port)"/>
    <param name="tf_frame_id" value="imu_link" />
    <param name="tf_parent_frame_id" value="imu_board_link" />
    <param name="broadcast_tf" value="true" />
    <!-- Remap imu data     -->
    <remap from="imu" to="imu/data"/>

    <remap from="joy" to="joy_teleop/joy" /> 
  </node>

  <!-- Launch IMU coordinate transform in base link -->
  <!--<node pkg="imu_transformer" type="imu_transformer_node" name="imu_transformer" output="screen">
    <remap from="imu_in/data"  to="imu/data"/>
    <remap from="imu_out/data" to="imu"/>
  </node>
  -->
  
</launch>
